<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from xacro-xARM.urdf                | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="xARM">
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="green">
    <color rgba="0 0.6 0 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  
  
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.155 0.14 0.035"/>
      </geometry>
      <material name="blue"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.155 0.14 0.035"/>
      </geometry>
    </collision>
  </link>

  <link name="rotary_table">
    <visual>
      <geometry>
        <cylinder length="0.01" radius="0.06"/>
      </geometry>
      <material name="white"/>
      <origin xyz="0 0 0"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.01" radius="0.06"/>
      </geometry>      
    </collision>
  </link>

  <joint name="base_to_rotary_table" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="2.0" lower="-1.5708" upper="1.5708" velocity="0.5"/>
    <parent link="base_link"/>
    <child link="rotary_table"/>
    <origin xyz="0 0 0.035"/>
    <!--origin rpy="0 0 1" xyz="0 0 0.0175"/-->
  </joint>

  <link name="servo5_link">
    <visual>
      <geometry>
        <box size="0.04402 0.027 0.028"/>
      </geometry>
      <material name="green"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.04402 0.027 0.028"/>
      </geometry>
    </collision>
  </link>

  <joint name="servo5_link_to_rotary_table" type="fixed">
    <parent link="rotary_table"/>
    <child link="servo5_link"/>
    <origin xyz="0 0 0.014"/>
  </joint>

  <link name="armlink1_part1">
    <visual>
      <geometry>
        <box size="0.033 0.003 0.055"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0.0015 0.0125"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.033 0.003 0.055"/>
      </geometry>
      <origin xyz="0 0.0015 0.0125"/>
    </collision>
  </link>

  <joint name="servo5_link_to_armlink1" type="revolute">
    <limit effort="2.0" lower="-1.5708" upper="1.5708" velocity="0.5"/>
    <parent link="servo5_link"/>
    <child link="armlink1_part1"/>
    <origin xyz="0.012 0.018 0.005"/>
    <axis xyz="0 -1 0"/>
  </joint>

  <link name="armlink1_part2">
    <visual>
      <geometry>
        <box size="0.033 0.042 0.003"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 -0.021 0"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.033 0.042 0.003"/>
      </geometry>
      <origin xyz="0 -0.021 0"/>
    </collision>
  </link>




  <joint name="armlink1_part1_to_part2" type="fixed">
    <parent link="armlink1_part1"/>
    <child link="armlink1_part2"/>
    <origin xyz="0 0.003 0.0385"/>
  </joint>

  <link name="armlink1_part3">
    <visual>
      <geometry>
        <box size="0.033 0.003 0.055"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0.0015 -0.0275"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.033 0.003 0.055"/>
      </geometry>
      <origin xyz="0 0.0015 -0.0275"/>
    </collision>
  </link>

  <joint name="armlink1_part2_to_part3" type="fixed">
    <parent link="armlink1_part2"/>
    <child link="armlink1_part3"/>
    <origin xyz="0 -0.042 0.0015"/>
  </joint>

  <link name="armlink1_part4">
    <visual>
      <geometry>
        <cylinder length="0.03" radius="0.0109"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0 0.015"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.03" radius="0.0109"/>
      </geometry>
      <origin xyz="0 0 0.015"/>
    </collision>
  </link>

  <joint name="armlink1_part2_to_part4" type="fixed">
    <parent link="armlink1_part2"/>
    <child link="armlink1_part4"/>
    <origin xyz="0 -0.021 0.0015"/>
  </joint>

  <link name="armlink1_part5">
    <visual>
      <geometry>
        <box size="0.025 0.042 0.003"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0 0.0015"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.025 0.042 0.003"/>
      </geometry>
      <origin xyz="0 0 0.0015"/>
    </collision>
  </link>

  <joint name="armlink1_part4_to_part5" type="fixed">
    <parent link="armlink1_part4"/>
    <child link="armlink1_part5"/>
    <origin xyz="0 0 0.03"/>
  </joint>

  <link name="armlink1_part6">
    <visual>
      <geometry>
        <box size="0.025 0.003 0.045"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0 0.0225"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.025 0.003 0.045"/>
      </geometry>
      <origin xyz="0 0 0.0225"/>
    </collision>
  </link>

  <joint name="armlink1_part5_to_part6" type="fixed">
    <parent link="armlink1_part5"/>
    <child link="armlink1_part6"/>
    <origin xyz="0 0.0195 0"/>
  </joint>

  <link name="armlink1_part7">
    <visual>
      <geometry>
        <box size="0.025 0.003 0.045"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0 0.0225"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.025 0.003 0.045"/>
      </geometry>
      <origin xyz="0 0 0.0225"/>
    </collision>
  </link>

  <joint name="armlink1_part5_to_part7" type="fixed">
    <parent link="armlink1_part5"/>
    <child link="armlink1_part7"/>
    <origin xyz="0 -0.0195 0"/>
  </joint>

  <link name="servo4_link">
    <visual>
      <geometry>
        <box size="0.02292 0.027 0.04822"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0.0135 0.01174"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.02292 0.027 0.04822"/>
      </geometry>
      <origin xyz="0 0.0135 0.01174"/>
    </collision>
  </link>

  <joint name="armlink1_to_servo4" type="revolute">
    <limit effort="2.0" lower="-2.09439510239" upper="2.09439510239" velocity="0.5"/>
    <parent link="armlink1_part7"/>
    <child link="servo4_link"/>
    <origin xyz="0 0.006 0.02942"/>
    <axis xyz="0 -1 0"/>
  </joint>

  <link name="armlink2">
    <visual>
      <geometry>
        <cylinder length="0.021" radius="0.011435"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0 0.0105"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.021" radius="0.011435"/>
      </geometry>
      <origin xyz="0 0 0.0105"/>
    </collision>
  </link>

  <joint name="servo4_to_armlink2" type="fixed">
    <parent link="servo4_link"/>
    <child link="armlink2"/>
    <origin xyz="0 0.0135 0.03585"/>
  </joint>

  <link name="servo3_link">
    <visual>
      <geometry>
        <box size="0.02292 0.027 0.04822"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0 0.02411"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.02292 0.027 0.04822"/>
      </geometry>
      <origin xyz="0 0 0.02411"/>
    </collision>
  </link>

  <joint name="armlink2_to_servo3" type="fixed">
    <parent link="armlink2"/>
    <child link="servo3_link"/>
    <origin xyz="0 0 0.021"/>
  </joint>

  <link name="armlink3_part1">
    <visual>
      <geometry>
        <box size="0.04698 0.003 0.06371"/>
      </geometry>
      <material name="green"/>
      <origin xyz="-0.01013 0.0015 0.018495"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.04698 0.003 0.06371"/>
      </geometry>
      <origin xyz="-0.01013 0.0015 0.018495"/>
    </collision>
  </link>

  <joint name="servo3_to_armlink3_part1" type="revolute">
    <limit effort="2.0" lower="-2.09439510239" upper="2.09439510239" velocity="0.5"/>
    <parent link="servo3_link"/>
    <child link="armlink3_part1"/>
    <origin xyz="0 0.02125 0.03468"/>
    <axis xyz="0 1 0"/>
  </joint>

  <link name="servo2_link">
    <visual>
      <geometry>
        <box size="0.04698 0.04250 0.02738"/>
      </geometry>
      <material name="green"/>
      <origin xyz="-0.02349 -0.02125 -0.01369"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.04698 0.04250 0.02738"/>
      </geometry>
      <origin xyz="-0.02349 -0.02125 -0.01369"/>
    </collision>
  </link>

  <joint name="armlink3_part1_to_servo2" type="fixed">
    <parent link="armlink3_part1"/>
    <child link="servo2_link"/>
    <origin xyz="0.01336 0 0.05035"/>
  </joint>

  <link name="armlink3_part2">
    <visual>
      <geometry>
        <box size="0.04698 0.003 0.06371"/>
      </geometry>
      <material name="green"/>
      <origin xyz="-0.02349 -0.0015 -0.031855"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.04698 0.003 0.06371"/>
      </geometry>
      <origin xyz="-0.02349 -0.0015 -0.031855"/>
    </collision>
  </link>

  <joint name="servo2_to_armlink3_part2" type="fixed">
    <parent link="servo2_link"/>
    <child link="armlink3_part2"/>
    <origin xyz="0 -0.0425 0"/>
  </joint>

  <link name="armlink4">
    <visual>
      <geometry>
        <cylinder length="0.022" radius="0.01"/>
      </geometry>
      <material name="green"/>
      <origin xyz="0 0 0.011"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.022" radius="0.01"/>
      </geometry>
      <origin xyz="0 0 0.011"/>
    </collision>
  </link>

  <joint name="servo2_to_armlink4" type="revolute">
    <limit effort="2.0" lower="-2.09439510239" upper="2.09439510239" velocity="0.5"/>
    <parent link="servo2_link"/>
    <child link="armlink4"/>
    <origin xyz="-0.013 -0.02125 0"/>
    <axis xyz="0 0 1"/>
  </joint>

  <link name="servo1_link">
    <visual>
      <geometry>
        <box size="0.05137 0.05682 0.02047"/>
      </geometry>
      <material name="green"/>
      <origin xyz="-0.015685 0 0.010235"/>
    </visual>
    <collision>
      <geometry>
        <box size="0.05137 0.05682 0.02047"/>
      </geometry>
      <origin xyz="-0.015685 0 0.010235"/>
    </collision>
  </link>

  <joint name="armlink4_to_servo1" type="fixed">
    <parent link="armlink4"/>
    <child link="servo1_link"/>
    <origin xyz="0 0 0.022"/>
  </joint>

  <joint name="left_gripper_joint" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="2.0" lower="-2.09439510239" upper="2.09439510239" velocity="0.5"/>
    <origin xyz="0.01 0 0.02047"/>
    <parent link="servo1_link"/>
    <child link="left_gripper"/>
  </joint>

  <link name="left_gripper">
    <visual>
      <origin xyz="0.0075 0 0.035"/>
      <geometry>
        <box size="0.015 0.015 0.1"/>
      </geometry>
      <material name="green"/>
    </visual>
    <collision>
      <origin xyz="0.0075 0 0.035"/>
      <geometry>
        <box size="0.015 0.015 0.1"/>
      </geometry>
    </collision>
  </link>
</robot>
